contracts/test-contracts/MockContract.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+contract MockContract {
+    event Called(uint256 value, bytes data);
+
+    struct CallResult {
+        bytes calldata_;
```

Comment:
it is a nit, but could you please rename it? I understand that the word "calldata" wouldn't work, maybe better something like "input"? Having an underscore at the end does not look aesthetic : )